src/buildstream/downloadablefilesource.py

```diff
@@ -235,8 +235,12 @@ def _ensure_mirror(self, activity_name: str):
                 return self.ref
 
             # Make sure url-specific mirror dir exists.
-            if not os.path.isdir(self._mirror_dir):
-                os.makedirs(self._mirror_dir)
+            try:
+                os.makedirs(self._mirror_dir, exist_ok=True)
```

Comment:
How is this even possible to happen?

Reply:
Is this basically a "plugin completely broke backwards-compat" case?

Reply:
This can happen whenever you have 2 separate elements which download the same downloadable file source.

The later call to `os.rename()` below guarantees the required atomicity, but the call to `os.makedirs()` will collide in the case that both sources try to create the directory after both having passed the `not os.path.isdir(self._mirror_dir)` check.